### PR TITLE
Added heuristics about which file to update in DependencyManager

### PR DESCRIPTION
**Pull request by andrecsilva**
- Also added a report listing if dependencies were added successfully or not.
- Added some robustness to RequirementsTxtWriter (this is mostly a band-aid and needs proper, more general, handling).

### CODE REVIEW

File: integration_tests/test_dependency_manager.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import os
+import subprocess
+import shutil
+import pytest
+from integration_tests.base_test import SAMPLES_DIR, CleanRepoMixin
+
+
+class TestDependencyManager(CleanRepoMixin):
```
Comment (drdavella) [inline integration_tests/test_dependency_manager.py:9]: Shouldn't we have test cases here for updating `pyproject.toml` and `setup.py` as well? In those cases we could just add some simple content to those temporary files within individual test cases.

File: src/codemodder/context.py [hunk 2]
```diff
@@ -85,28 +86,41 @@ def get_failed_files(self):
...
         # pylint: disable-next=cyclic-import
         from codemodder.dependency_management import DependencyManager
 
-        dm = DependencyManager(dependencies_store, self.directory)
-        if (changeset := dm.write(list(dependencies), self.dry_run)) is not None:
-            self.add_results(codemod_id, [changeset])
+        for package_store in store_list:
+            dm = DependencyManager(package_store, self.directory)
+            if (changeset := dm.write(list(dependencies), self.dry_run)) is not None:
+                self.add_results(codemod_id, [changeset])
```
Comment (drdavella) [inline src/codemodder/context.py:118]: Maybe a debug log here saying which file is being written?

**Review by drdavella — CHANGES_REQUESTED** [review]
Apologies for the delayed review. It looks good overall but I think we might need some more tests to enumerate a few more cases.

File: src/codemodder/context.py (hunk 2 above)
Comment (andrecsilva) [reply src/codemodder/context.py:118]: This information is being store in the `record` variable. The debug message with the file the dependency has been written in is in the new `record_dependency_update` function at `codemodder.py`

**Review by drdavella — APPROVED** [review]

**Pull request merged**